packages/commonjs/src/base-cjs-module-system.ts

```diff
@@ -27,15 +29,24 @@ export interface IBaseModuleSystemOptions {
    * `undefined` - couldn't resolve request.
    */
   resolveFrom(contextPath: string, request: string, requestOrigin?: string): string | false | undefined;
+
+  wrapRequire?: (require: RequireCall, loadedModules: Map<string, IModule>) => RequireCall;
 }
 
+const defaultWrapRequire =
+  (req: (modulePath: string | false) => unknown): ((path: string | false) => unknown) =>
+  (path) =>
+    req(path);
+
 export function createBaseCjsModuleSystem(options: IBaseModuleSystemOptions): ICommonJsModuleSystem {
-  const { resolveFrom, dirname, readFileSync, globals = {} } = options;
+  const { resolveFrom, dirname, readFileSync, globals = {}, wrapRequire = defaultWrapRequire } = options;
 
   const loadedModules = new Map<string, IModule>();
 
+  const wrappedRequireModule = wrapRequire(requireModule, loadedModules);
```

Comment:
We don't want to wrap if we don't need it's just en extra function call and worth the less nice code